src/cryptography/hazmat/primitives/asymmetric/rsa.py

```diff
@@ -190,6 +190,38 @@ def rsa_crt_dmq1(private_exponent: int, q: int) -> int:
     return private_exponent % (q - 1)
 
 
+def rsa_recover_private_exponent(e: int, p: int, q: int) -> int:
+    """
+    Compute the RSA private_exponent (d) given the public exponent (e)
+    and the RSA primes p and q.
+
+    This uses the Carmichael totient function to generate the
+    smallest possible working value of the private exponent.
+    """
+    # This lambda_n is the Carmichael totient function.
+    # The original RSA paper uses the Euler totient function
+    # here: phi_n = (p - 1) * (q - 1)
+    # Either version of the private exponent will work, but the
+    # one generated by the older formulation may be larger
+    # than necessary. (lambda_n always divides phi_n)
+    lambda_n = (p - 1) * (q - 1) // gcd(p - 1, q - 1)
+    return _modinv(e, lambda_n)
+
+
+def _rsa_recover_euler_private_exponent(e: int, p: int, q: int) -> int:
```

Comment:
Let's move this implementation into the test package since it has no use outside of the tests.

Reply:
Done in 5a58cb2cca2f302e28195b37b01a1f6014611af6

Reply:
Done in https://github.com/pyca/cryptography/commit/5a58cb2cca2f302e28195b37b01a1f6014611af6